app/src/main/java/com/keylesspalace/tusky/fragment/SFragment.java

```diff
@@ -342,4 +356,31 @@ private void showOpenAsDialog(String statusUrl, CharSequence dialogTitle) {
         BaseActivity activity = (BaseActivity)getActivity();
         activity.showAccountChooserDialog(dialogTitle, false, account -> openAsAccount(statusUrl, account));
     }
+
+    private void downloadAllMedia(Status status) {
+        pendingDownloadStatus = null;
+        Toast.makeText(getContext(), R.string.downloading_media, Toast.LENGTH_SHORT).show();
+        for(Attachment attachment: status.getAttachments()) {
+            String url = attachment.getUrl();
+            Uri uri = Uri.parse(url);
+            String filename = uri.getLastPathSegment();
+
+            DownloadManager downloadManager = (DownloadManager)getActivity().getSystemService(Context.DOWNLOAD_SERVICE);
+            DownloadManager.Request request = new DownloadManager.Request(uri);
+            request.setDestinationInExternalPublicDir(Environment.DIRECTORY_DOWNLOADS, filename);
+            downloadManager.enqueue(request);
+        }
+    }
+
+    private void requestDownloadAllMedia(Status status) {
+        pendingDownloadStatus = status;
+        String[] permissions = new String[]{ Manifest.permission.WRITE_EXTERNAL_STORAGE };
+        ((BaseActivity)getActivity()).requestPermissions(permissions, Build.VERSION_CODES.M, (permissions1, grantResults) -> {
```

Comment:
if you don't use lambda arg, I would rename it to `__` but that's not important obv

Reply:
Aha, so the interface being in kotlin is why I couldn't lambdify the kotlin call